server/game/gameSystems/SelectCardSystem.ts

```diff
@@ -78,8 +78,9 @@ export class SelectCardSystem<TContext extends AbilityContext = AbilityContext>
         if (properties.mode === TargetMode.UpTo || properties.mode === TargetMode.UpToVariable) {
             properties.canChooseNoCards = properties.canChooseNoCards ?? true;
         } else {
-            // should not use this property if mode is not UpTo(Variable)
-            properties.canChooseNoCards = false;
+            if (properties.canChooseNoCards === true) {
```

Comment:
Should be `properties.canChooseNoCards != null` to check if they put any value at all